Replace user-facing Blacklist/Whitelist with Blocklist/Allowlist

bradyjoh opened the pull request:
- [x] You have read through `CONTRIBUTING.md`
- [x] Your changes are not possible to do through a plugin and relevant
  to a large audience (ideally all users of OctoPrint)
- [x] If your changes are large or otherwise disruptive: You have
  made sure your changes don't interfere with current development by
  talking it through with the maintainers, e.g. through a
  Brainstorming ticket
- [x] Your PR targets OctoPrint's `dev` branch
- [x] Your PR was opened from a custom branch on your repository
  (no PRs from your version of `main`, `bugfix`, `next` or `dev`
  please), e.g. `wip/my_new_feature` or `wip/my_bugfix`
- [x] Your PR only contains relevant changes: no unrelated files,
  no dead code, ideally only one commit - rebase and squash your PR
  if necessary!
- [x] If your changes include style sheets: You have modified the
  `.less` source files, not the `.css` files (those are generated
  with `lessc`)
- [x] You have tested your changes (please state how!) - ideally you
  have added unit tests
- [x] You have run the existing unit tests against your changes and
  nothing broke (`pytest`)
- [x] You have run the included `pre-commit` suite against your changes
  and nothing broke (`pre-commit run --all-files`)
- [x] You have added yourself to the `AUTHORS.md` file :)


#### What does this PR do and why is it necessary?

This PR replaces string instances of Blacklist and Whitelist (and their derivatives black list and white list, CAPS and no caps) with a more inclusive alternative, Blocklist and Allowlist. More discussion in the ticket https://github.com/OctoPrint/OctoPrint/issues/5207

#### How was it tested? How can it be tested by the reviewer?

I ran `pytest` locally after following https://github.com/OctoPrint/OctoPrint/blob/main/CONTRIBUTING.md#setting-up-a-development-environment and https://docs.octoprint.org/en/main/development/environment.html. `1096 passed, 47 warnings in 64.99s (0:01:04)`

I also manually ran the  pre-commit check suite with `pre-commit run --hook-stage manual --all-files` and those all passed.

I also ran the octoprint server locally on my macbook with `octoprint serve --port 12345` and tried to go through the setup wizard. I also went into the settings after the setup wizard to confirm the UI showed the Blocklist language

#### Was any kind of genAI (ChatGPT, Copilot etc) involved in creating this PR?

None, I used PyCharm to find and replace all the instances by hand

#### Any background context you want to provide?

N/A

#### What are the relevant tickets if any?

https://github.com/OctoPrint/OctoPrint/issues/5207

#### Screenshots (if appropriate)

<img width="763" height="504" alt="Screenshot 2025-11-24 at 12 20 38" src="https://github.com/user-attachments/assets/0bb35c5f-3563-4b55-8031-a783bb6e9238" />
<img width="310" height="359" alt="Screenshot 2025-11-24 at 12 20 26" src="https://github.com/user-attachments/assets/ea14cc7c-c894-4e5e-a4b8-bf57433b3334" />
<img width="953" height="364" alt="Screenshot 2025-11-24 at 12 32 45" src="https://github.com/user-attachments/assets/059c2fd6-5b66-49ca-9403-82a40c7b3f90" />
<img width="884" height="144" alt="Screenshot 2025-11-24 at 12 30 50" src="https://github.com/user-attachments/assets/2c2e0c82-cc3a-41d4-99a1-8f2d24823185" />


#### Further notes

Thank you so much for setting up this project to make contributions easy ❤️

foosel commented on .pre-commit-config.yaml:
```diff
@@ -1,5 +1,5 @@
 default_language_version:
-  python: python3.12
+  python: python3.13
```
This line as well as the default python version change in the GitHub Action workflow should be removed from the commit.

The reason you ran into errors on your local run of `pre-commit` was that you didn't have Python 3.12 available on your system. Mea culpa - I need to add this to the dev guide. I think something like `uv python install cpython3.12` should fix this already I think (given `uv` is installed - if not, `pip install uv` should do the trick)

bradyjoh replied:
I ended up using https://github.com/pyenv/pyenv to install python 3.12 specifically for this directory, I couldn't get uv working the way I expected. My PATH and build environment seems really messed up! 

Anyway, with `pyenv install 3.12` I was able to keep a 3.12 copy local in the OctoPrint directory, and then the pre-commit worked as expected

foosel commented:
I'll do my best to give this a review this week, but I have a bunch of other stuff going on.

One thing I already saw though, you changed wording in a bundled javascript library (`pako.js`). I'd prefer to avoid that.

Also, if the goal here is to stick to user visible wording (which I understood you to want to go with), I'd actually remove all the comment changes too, it should only be stuff that actually ends on the UI or in the log for now, so translation files, Jinja2 templates, JS files, logging. Alternatively, we can also go with everything, but then we are back on full blown code changes + setting migration. I'm not against the latter as already mentioned, we just should decide what this PR's goal is and focus on that.

bradyjoh commented:
Thanks! No rush at all, I'll make your requested changes soon and send an update once that's done 🦖

bradyjoh commented:
OK! I have made adjustments based on your feedback and removed many of the changes. I removed the .POT changes because I learned those files are autogenerated from .PO files? I hope that's OK. Please comment on anything you think still needs to be removed!

I reran `pytest` and confirmed it looks good. I see the GH Actions/Checks have passed. Python is back to 3.12.

Also ran OctoPrint from the dev environment and quickly looked in the UI to make sure it was as expected:

<img width="663" height="353" alt="image" src="https://github.com/user-attachments/assets/ecfdb3b7-a126-4443-9cac-73a2bd727e7e" />
<img width="716" height="438" alt="image" src="https://github.com/user-attachments/assets/3e06c401-960e-4907-a59c-7a5dd17941a9" />


Thank you for being patient with me as I learn about contributing to this project 😁

foosel commented:
Looking good, just merged it, thank you for this!
